buildSrc/src/main/kotlin/codegen/GenerateKotlinDependencyExtensions.kt

```diff
@@ -74,10 +74,10 @@ fun RepositoryHandler.gradleScriptKotlin(): ArtifactRepository =
  * Builds the dependency notation for the named Kotlin [module] at the given [version].
  *
  * @param module simple name of the Kotlin module, for example "reflect".
- * @param version optional desired version, null implies [embeddedKotlinVersion].
+ * @param version optional desired version, unspecified if null.
  */
 fun DependencyHandler.kotlin(module: String, version: String? = null): Any =
-    "org.jetbrains.kotlin:kotlin-${'$'}module:${'$'}{version ?: embeddedKotlinVersion}"
+    "org.jetbrains.kotlin:kotlin-${'$'}module${'$'}{if(version == null) "" else ":${'$'}version"}"
```

Comment:
:0: I would express this using `?let` and null-coalescing `version?.let { ":${'$'}version" } ?: ""`

Reply:
👍 Done